src/route/builder.rs

```diff
@@ -168,6 +168,13 @@ impl RouteMessageBuilder<Ipv4Addr> {
             .push(RouteAttribute::Gateway(RouteAddress::Inet(addr)));
         self
     }
+
+    pub fn gateway6(mut self, addr: Ipv6Addr) -> Self {
```

Comment:
Since we already patched `gateway()` to support assigning ipv6 gateway to a ipv4 route, we do not need this function anymore.

If you precise control, maybe change this function to `pub fn via(mut self, via: RouteVia) -> Self`.

Reply:
Yes, anyone who wants an IPv4 route with an IPv6 nexthop should probably be using `RouteMessageBuilder<IpAddr>`. I think this method could be dropped.

Reply:
Sorry for the late reply. Changed the function name to `via`, as IMO it would be less consistent in regards of functionality between the `Ipv4Addr` and `IpAddr` builders, if removing this completely.